chainer/links/normalization/batch_normalization.py

```diff
@@ -168,3 +185,11 @@ def start_finetuning(self):
 
         """
         self.N = 0
+
+
+def _init_array(initializer, default_value, size, dtype):
+    if initializer is None:
+        initializer = default_value
+    initializer = initializers._get_initializer(initializer)
+    initializer.dtype = dtype
```

Comment:
`dtype` attribute is a part of the initializer interface (for example, see https://docs.chainer.org/en/stable/reference/generated/chainer.initializers.generate_array.html#chainer.initializers.generate_array). Maybe it's better to fix the documentation for initializers? (Btw, it does not raise `AttributeError` in general to set any attributes to callables; even function (including lambda) can have a custom attribute)

Reply:
Ah, I have missed that statement in `initializers.generate_array`. I understand.

(I thought it would raise `AttibuteError` because `a = object(); a.b = 3` did. It seems `object` is different...)

Reply:
But previously, `initializer.dtype` is not particularly meant to be writable (both in existing code and documentation).

Rather than forcing it to be writable, how about adding `dtype` (`default_dtype`?) argument to `initializers.generate_array()`?

Reply:
That sounds good. I will add `dtype` option to `generate_array`.

Reply:
fixed